src/components/ButtonDropdownSelect/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {useState} from 'react';
+import { Box, ListItemIcon, ListItemText, Menu, MenuItem, MenuList, Typography } from '@mui/material';
+import { DropdownItem } from '../types';
+import { IconName } from '../Icon/icons';
+import Button from '../Button/Button';
+import Icon from '../Icon/Icon';
+import './styles.scss';
+
+type ButtonDropdownSelectProps = {
+  id: string;
+  options: DropdownItem[];
+  initialSelection: DropdownItem[];
+  onChange: (selection: DropdownItem[]) => void;
+  buttonIcon?: IconName;
+  buttonLabel?: string;
+  menuAlign?: 'left' | 'center' | 'right';
+  buttonStyle?: 'terraware' | 'plain';
+  disabled?: boolean;
+};
+
+export default function ButtonDropdownSelect(props: ButtonDropdownSelectProps): JSX.Element {
+  const {id, options, initialSelection, onChange, buttonIcon, buttonLabel, menuAlign, buttonStyle, disabled} = props;
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event?: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event?.currentTarget ?? null);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const [selection, setSelection] = useState(initialSelection);
+  const addToSelection = (value: DropdownItem) => {
+    const newSelection = [...selection];
+    newSelection.push(value);
+    setSelection(newSelection);
+    onChange(newSelection);
+  };
+  const removeFromSelection = (index: number) => {
+    const newSelection = [...selection];
+    newSelection.splice(index, 1);
+    setSelection(newSelection);
+    onChange(newSelection);
+  };
+
+  const iconSpacer = () => {
+    return <div className='button-dropdown-select__checkmark-spacer'/>;
+  };
+
+  return <>
+    {buttonStyle === 'terraware' ? (
+      <Button onClick={handleClick} label={buttonLabel} icon={buttonIcon} disabled={disabled}/>
+    ) : (
+      <button onClick={handleClick} className='button-dropdown-select__plain-button' disabled={disabled}>
+        {buttonIcon && <Icon name={buttonIcon} size='small' className='plain-button__icon'/>}
+        {buttonLabel && <Typography className='plain-button__label'>{buttonLabel}</Typography>}
+      </button>
+    )}
+    <Menu
```

Comment:
Thoughts on using the existing [popover component](https://github.com/terraware/web-components/blob/main/src/components/PopoverMenu/Popover.tsx)? We could enhance that with a an optional renderer prop (for the label/check, etc.).
If it's not feasible, that's fine.

Reply:
Okay, I changed this and I like it much more. I kept the component here, feeling it is generic enough for general use. The Popover component was used, with an enhancement for custom item rendering as you suggested.

Reply:
I'll create the MapLayerSelect as a component in terraware-web, built on this

Reply:
> Okay, I changed this and I like it much more. I kept the component here, feeling it is generic enough for general use. The Popover component was used, with an enhancement for custom item rendering as you suggested.

Thanks for following up @charlie-tf - I will take a look